data/data-pipeline/data_pipeline/etl/sources/cdc_svi_index/etl.py

```diff
@@ -0,0 +1,114 @@
+import pandas as pd
+
+from data_pipeline.etl.base import ExtractTransformLoad
+from data_pipeline.utils import get_module_logger
+from data_pipeline.score import field_names
+from data_pipeline.config import settings
+
+
+logger = get_module_logger(__name__)
+
+
+class CDCSVIIndex(ExtractTransformLoad):
+    """CDC SVI Index class ingests 2018 dataset located
+    here: https://www.atsdr.cdc.gov/placeandhealth/svi/index.html
+    Please see the README in this module for further details.
+    """
+
+    def __init__(self):
+        self.OUTPUT_PATH = self.DATA_PATH / "dataset" / "cdc_svi_index"
+
+        self.CDC_SVI_INDEX_S3_URL = (
+            settings.AWS_JUSTICE40_DATASOURCES_URL + "/SVI2018_US.zip"
+        )
+
+        self.CDC_RPL_THEMES_THRESHOLD = 0.90
+
+        self.CDC_SVI_INDEX_ZIP_FILE_DIR = self.TMP_PATH / "cdc_svi_index"
+
+        self.CDC_SVI_INDEX_TRACTS_FIPS_CODE = "FIPS"
+
+        self.COLUMNS_TO_KEEP = [
+            self.GEOID_TRACT_FIELD_NAME,
+            field_names.CDC_SVI_INDEX_SE_THEME_PERCENTILE_FIELD,  # SE prefix references "socioeconomic"
+            field_names.CDC_SVI_INDEX_HOUSEHOLD_THEME_PERCENTILE_COMPOSITION_FIELD,
+            field_names.CDC_SVI_INDEX_LANGUAGE_THEME_PERCENTILE_FIELD,
+            field_names.CDC_SVI_INDEX_HOUSING_TRANSPORTATION_PERCENTILE_FIELD,
+            field_names.CDC_SVI_INDEX_RPL_THEMES_OVERALL_PERCENTILE_FIELD,
+            # Derived columns
+            field_names.CDC_SVI_INDEX_THEMES_PRIORITY_COMMUNITY,
+        ]
+
+        self.df: pd.DataFrame
+
+    def extract(self) -> None:
+        logger.info("Extracting 43 MB CDC SVI INDEX")
+        super().extract(
+            self.CDC_SVI_INDEX_S3_URL,
+            self.CDC_SVI_INDEX_ZIP_FILE_DIR,
+        )
+
+    def transform(self) -> None:
+        logger.info("Starting CDC SVI INDEX transform")
+
+        # New file name
+        tmp_svi_csv_file_path = (
+            self.CDC_SVI_INDEX_ZIP_FILE_DIR / "SVI2018_US.csv"
+        )
+
+        self.df = pd.read_csv(
+            filepath_or_buffer=tmp_svi_csv_file_path,
+            dtype={self.CDC_SVI_INDEX_TRACTS_FIPS_CODE: str},
+        )
+
+        # Note: In this dataset all US census tracts are ranked against one another.
+        # Puerto Rico is not included in this dataset
+        self.df.rename(
+            columns={
+                self.CDC_SVI_INDEX_TRACTS_FIPS_CODE: self.GEOID_TRACT_FIELD_NAME,
```

Comment:
Based on the documentation, it looks like some of these fields will have -999 in them – could you check for that and replace them with NaNs if they exist?

Reply:
Yes, I made note of this in the README as well. Yes, I can impute those with `np.nan`